### ISSUE TITLE
.SendActivity() takes Activity and not IActivity; IMessageActivity objects are unusable (ie: MessageFactory outputs)

### ISSUE THREAD

**Issue by brandonh-msft** [Microsoft/botbuilder-dotnet#282]
```csharp
var m = MessageFactory.Attachment(default(Attachment));
await context.SendActivity(m);
```
no longer works in latest master branch because the new `SendActivity` method takes `params Activity[]` instead of `IActivity` or something to which `IMessageActivity` objects are implicitly convertible

**Comment by brandonh-msft** [comment]
cc @tomlm @cleemullins @cmayomsft

**Comment by cleemullins** [comment]
For right now, just cast to Activity.

Need to figure out a cleaner answer.

________________________________

**Comment by cleemullins** [comment]
Fixed in PR #284. Once that's approved, I'll merge it in.

**Comment by cleemullins** [comment]
The relationship between Activity, IActivity, IMessageACtivity and other similar interfaces are strange. I would love ideas from folks on how to normalize this better. What's there now is a total carryover from the previous Schema / Connector libraries.

**Comment by drub0y** [comment]
I have some thoughts on this.... especially as it relates to the patterns people have to use when handling activities... the whole AsXXX feels non-idiomatic to a typed language where I should just be able to test the actual type with a language feature (especially w/C# 7 where we have gained more pattern matching features in the language). Also the cost of the implementation is quite high (re-deserializing IIRC).

I believe that the host should be responsible for deserializing from the transport to the exact runtime type that the activity is (e.g. `MessageActivity` instead of just `Activity`) and then it can drop it through the pipeline as `[I]Activity` and downstream a pieces of middleware or the bot itself can now just use idiomatic language features to check the run time type.

I imagine this should go through a proper DCR style discussion in its own issue though, right?

**Comment by Stevenic** [comment]
I personally believe that we should drop the `IXxxxActivity` parts all together as this is not in parity with the other languages.  Unless we change our swagger to generate these interfaces we just shouldn't use them. I took this approach in the v3 Node SDK and have spent the last 2 years having to manually keep them updated with changes. That way lies madness...

To be at parity with the other languages you should just pass around an `IActivity` everywhere and the developer needs to know what fields are used for which type. Don't like that? Fix the swagger...

**Comment by drub0y** [comment]
I agree with that @Stevenic. I have yet to see anything that necessitates `IActitivity`. For data classes like the activities there's very little benefit to an interface as there is no _behavior_ that would ever need to be mocked and for testing purposes you can always instantiate any one of the classes and "fill them out" with the test data you want.

**Comment by cleemullins** [comment]
The issue here, as defined by the title, is resolved by #284. If we want to evaluate the "IActivity" vs "Activity" vs "Multiple Interfaces" I think that's a great discussion but should be in a dedicated topic. 

Closing this out, as it was a bug. :)
